### PR TITLE
Fix private method hidden by ancestor class attribute

### LINKED ISSUE #3936: E0202: (method-hidden) false positive when mixed member/method
Generates false positive:
```python
class Parent:
    def __init__(self):
        self.__check_should_start = None

class Child(Parent):
    def __check_should_start(self):
        pass
```

works OK:
```python
class Parent:
    def __check_should_start(self):
        pass

class Child(Parent):
    def __check_should_start(self):
        pass
```

```
$ pylint-3 --version
pylint 2.4.4
astroid 2.3.3
Python 3.8.6 (default, Sep 25 2020, 00:00:00) 
[GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

### PR DESCRIPTION
**Pull request by tiagohonorato**
## Steps

- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description

A class defined private method won't be hidden by an ancestor class private attribute as the class will mangle the latter to avoid naming collisions. This fixes a false positive for ``method-hidden`` when using private attributes and methods.

Signed-off-by: Tiago Honorato <tiagohonorato1@gmail.com>

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |
|   | :sparkles: New feature |
|   | :hammer: Refactoring  |
|    | :scroll: Docs |

## Related Issue

Closes #3936

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you for working on this. Could you add another test with a protected attribute that check that the ``method-hidden`` is raised in this case ?

```python
            class Parent:
                def __init__(self):
                    self._protected = None
            class Child(Parent):
                def _protected(self): # [method-hidden]
                    pass
```

**Comment by tiagohonorato** [comment]
Hi @Pierre-Sassoulas, thank you for the suggestion. I will add that test!

**Review by Pierre-Sassoulas — APPROVED** [review]

File: ChangeLog [hunk 1]
```diff
@@ -27,6 +27,12 @@ Release date: TBA
 ..
   Put bug fixes that will be cherry-picked to latest major version here
 
+* Fix false positive for `method-hidden` when using private attribute and method
+
+  Closes #3936
+
+* Fix issue with Enums and `class-attribute-naming-style=snake_case`
+
```
Comment (Pierre-Sassoulas) [inline ChangeLog:35]:
```suggestion
* Fix false positive for ``method-hidden`` when using private attribute and method

  Closes #3936

* Fix issue with Enums and ``class-attribute-naming-style=snake_case``

```

File: ChangeLog [hunk 2]
```diff
@@ -27,6 +27,12 @@ Release date: TBA
 ..
   Put bug fixes that will be cherry-picked to latest major version here
 
+* Fix false positive for ``method-hidden`` when using private attribute and method
+
+  Closes #3936
+
+* Fix issue with Enums and ``class-attribute-naming-style=snake_case``
+
```
Comment (Pierre-Sassoulas) [inline ChangeLog:35]:
```suggestion

```

**Pull request merged**